diff --git a/Capacitor/Capacitor/Plugins/HttpRequestHandler.swift b/Capacitor/Capacitor/Plugins/HttpRequestHandler.swift
index 651c1b76..2230b32d 100644
--- a/Capacitor/Capacitor/Plugins/HttpRequestHandler.swift
+++ b/Capacitor/Capacitor/Plugins/HttpRequestHandler.swift
@@ -224,11 +224,26 @@ open class HttpRequestHandler {
                 call.reject(error.localizedDescription, (error as NSError).domain, error, nil)
                 return
             }
-
-            setCookiesFromResponse(response as! HTTPURLResponse, config)
-
-            let type = ResponseType(rawValue: responseType) ?? .default
-            call.resolve(self.buildResponse(data, response as! HTTPURLResponse, responseType: type))
+                                                          
+            if let response = response as? HTTPURLResponse {
+                setCookiesFromResponse(response, config)
+    
+                let type = ResponseType(rawValue: responseType) ?? .default
+                call.resolve(self.buildResponse(data, response, responseType: type))
+            } else if let response = response as? NSURLResponse {
+                // applicable to data: URI requests
+                var headers = [:] as [String: Any]
+                headers["Content-Type"] = response.mimeType
+                
+                var output = [:] as [String: Any]
+                output["status"] = 200
+                output["headers"] = headers
+                output["data"] = String(data: data!, encoding: .utf8)
+                call.resolve(output)
+            } else {
+                call.reject("Unknown response kind")
+                return
+            }
         }
 
         task.resume()
