diff --git a/capacitor/src/main/java/com/getcapacitor/Bridge.java b/capacitor/src/main/java/com/getcapacitor/Bridge.java
index 2a72a430..12720652 100644
--- a/capacitor/src/main/java/com/getcapacitor/Bridge.java
+++ b/capacitor/src/main/java/com/getcapacitor/Bridge.java
@@ -434,7 +434,7 @@ public class Bridge {
             PackageManager pm = getContext().getPackageManager();
             PackageInfo pInfo = InternalUtils.getPackageInfo(pm, getContext().getPackageName());
             versionCode = Integer.toString((int) PackageInfoCompat.getLongVersionCode(pInfo));
-            versionName = pInfo.versionName;
+            if (pInfo.versionName != null) versionName = pInfo.versionName;
         } catch (Exception ex) {
             Logger.error("Unable to get package info", ex);
         }
