diff --git a/capacitor/src/main/java/com/getcapacitor/util/WebColor.java b/capacitor/src/main/java/com/getcapacitor/util/WebColor.java
index e055021e..fe62763c 100644
--- a/capacitor/src/main/java/com/getcapacitor/util/WebColor.java
+++ b/capacitor/src/main/java/com/getcapacitor/util/WebColor.java
@@ -11,7 +11,7 @@ public class WebColor {
      */
     public static int parseColor(String colorString) {
         String formattedColor = colorString;
-        if (colorString.charAt(0) != '#') {
+        if (!colorString.isEmpty() && colorString.charAt(0) != '#') {
             formattedColor = "#" + formattedColor;
         }
 
