diff --git a/dist/index.cjs.js b/dist/index.cjs.js
index 34d371f..e1f2a3b 100644
--- a/dist/index.cjs.js
+++ b/dist/index.cjs.js
@@ -435,13 +435,16 @@
         let encodedValue;
         let item;
         if (Array.isArray(value)) {
+            if (value.length === 0) {
+                return accumulator;
+            }
             item = '';
             value.forEach((str) => {
                 encodedValue = shouldEncode ? encodeURIComponent(str) : str;
                 item += `${key}=${encodedValue}&`;
             });
             // last character will always be "&" so slice it off
-            item.slice(0, -1);
+            item = item.slice(0, -1);
         }
         else {
             encodedValue = shouldEncode ? encodeURIComponent(value) : value;
@@ -450,7 +453,7 @@
         return `${accumulator}&${item}`;
     }, '');
     // Remove initial "&" from the reduce
-    return output.substr(1);
+    return output.substring(1);
 };
 /**
  * Build the RequestInit object based on the options passed into the initial request
diff --git a/dist/index.js b/dist/index.js
index fb980f2..f4e5d6c 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -431,13 +431,16 @@
         let encodedValue;
         let item;
         if (Array.isArray(value)) {
+            if (value.length === 0) {
+                return accumulator;
+            }
             item = '';
             value.forEach((str) => {
                 encodedValue = shouldEncode ? encodeURIComponent(str) : str;
                 item += `${key}=${encodedValue}&`;
             });
             // last character will always be "&" so slice it off
-            item.slice(0, -1);
+            item = item.slice(0, -1);
         }
         else {
             encodedValue = shouldEncode ? encodeURIComponent(value) : value;
@@ -446,7 +449,7 @@
         return `${accumulator}&${item}`;
     }, '');
     // Remove initial "&" from the reduce
-    return output.substr(1);
+    return output.substring(1);
 };
 /**
  * Build the RequestInit object based on the options passed into the initial request
