diff --git a/capacitor/src/main/java/com/getcapacitor/Bridge.java b/capacitor/src/main/java/com/getcapacitor/Bridge.java
index 2a72a430..5cde1e4b 100644
--- a/capacitor/src/main/java/com/getcapacitor/Bridge.java
+++ b/capacitor/src/main/java/com/getcapacitor/Bridge.java
@@ -279,6 +279,9 @@ public class Bridge {
 
         webView.setWebChromeClient(new BridgeWebChromeClient(this));
         webView.setWebViewClient(this.webViewClient);
+        webView.getSettings().setBuiltInZoomControls(true);
+        webView.setInitialScale(100);
+        webView.setPadding(0, 0, 0, 0);
 
         if (config.isResolveServiceWorkerRequests()) {
             ServiceWorkerController swController = ServiceWorkerController.getInstance();
