diff --git a/Capacitor/Capacitor/CAPApplicationDelegateProxy.swift b/Capacitor/Capacitor/CAPApplicationDelegateProxy.swift
index 7f510bc5..7fe2bae7 100644
--- a/Capacitor/Capacitor/CAPApplicationDelegateProxy.swift
+++ b/Capacitor/Capacitor/CAPApplicationDelegateProxy.swift
@@ -10,7 +10,7 @@ public class ApplicationDelegateProxy: NSObject, UIApplicationDelegate {
         NotificationCenter.default.post(name: .capacitorOpenURL, object: [
             "url": url,
             "options": options
-        ])
+        ] as [String : Any])
         NotificationCenter.default.post(name: NSNotification.Name.CDVPluginHandleOpenURL, object: url)
         lastURL = url
         return true
